src/components/InjectedComponents/ImmersiveSetup/Setup-Stepper.tsx

```diff
@@ -0,0 +1,285 @@
+import { geti18nString } from '../../../utils/i18n'
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        root: {
+            width: '100%',
+            maxWidth: 350,
+        },
+        button: {
+            marginTop: theme.spacing(1),
+            marginRight: theme.spacing(1),
+        },
+        provePost: {
+            wordBreak: 'break-all',
+            padding: 6,
+            border: `1px solid ${theme.palette.divider}`,
+        },
+        emptyProvePost: {
+            padding: 6,
+            border: `1px solid ${theme.palette.error.main}`,
+        },
+        header: { cursor: 'move' },
+    }),
+)
+
+export enum ImmersiveSetupState {
+    ConfirmUsername,
+    PasteBio,
+    Finished,
+}
+export interface ImmersiveSetupStepperUIProps {
+    loadProfile(): Promise<void>
+    autoPasteProvePost(): Promise<void>
+    bioSet: 'detecting' | boolean
+    currentStep: ImmersiveSetupState
+    back(): void
+    next(): void
+    provePost: string
+    username: string
+    onUsernameChange: (newUserName: string) => void
+    onClose(): void
+}
+export function ImmersiveSetupStepperUI(props: ImmersiveSetupStepperUIProps) {
+    const classes = useStyles()
+    const steps = getSteps()
+    const activeStep = props.currentStep
+    const [, inputRef] = useCapturedInput(props.onUsernameChange)
+
+    // const backButton = (
+    //     <Button onClick={props.back} className={classes.button}>
+    //         Back
+    //     </Button>
+    // )
+    const actions = (
+        <div>
+            {/* TODO: Implement back(including side effects rollback) for step 1  */}
+            {/* {activeStep <= 1 ? null : backButton} */}
+            {getNextButton(activeStep)}
+        </div>
+    )
+    return (
+        <aside className={classes.root}>
+            <AppBar component="nav" position="static" className={classes.header}>
+                <Toolbar variant="dense">
+                    <Typography variant="h6">{geti18nString('immersive_setup_title')}</Typography>
+                    <div style={{ flex: 1 }} />
+                    <IconButton edge="end" color="inherit" onClick={props.onClose}>
+                        <CloseIcon />
+                    </IconButton>
+                </Toolbar>
+            </AppBar>
+            <Stepper activeStep={activeStep} orientation="vertical">
+                {steps.map((label, index) => (
+                    <Step key={index}>
+                        <StepLabel>{label}</StepLabel>
+                        <StepContent>
+                            {getStepContent(index)}
+                            {actions}
+                        </StepContent>
+                    </Step>
+                ))}
+            </Stepper>
+            {/* {activeStep === steps.length && <Typography>{getBioStatus()}</Typography>} */}
+        </aside>
+    )
+    /** TODO: this UI is unused currently. */
+    function getBioStatus() {
+        switch (props.bioSet) {
```

Comment:
Maybe use enumerate better than this manner.

Reply:
this UI is not used currently. I'll add a note on it.